api/types/authority.go

```diff
@@ -288,15 +288,64 @@ func (ca *CertAuthorityV2) SetSigningAlg(alg CertAuthoritySpecV2_SigningAlgType)
 	ca.Spec.SigningAlg = alg
 }
 
-func (ca *CertAuthorityV2) GetActiveKeys() CAKeySet { return ca.Spec.ActiveKeys }
+func (ca *CertAuthorityV2) GetActiveKeys() CAKeySet {
+	haveNewCAKeys := len(ca.Spec.ActiveKeys.SSH) > 0 || len(ca.Spec.ActiveKeys.TLS) > 0 || len(ca.Spec.ActiveKeys.JWT) > 0
+	if haveNewCAKeys {
+		return ca.Spec.ActiveKeys
+	}
+	// fall back to old schema
+	var keySet CAKeySet
+	if len(ca.Spec.CheckingKeys) > 0 {
+		kp := &SSHKeyPair{
+			PrivateKeyType: PrivateKeyType_RAW,
+			PublicKey:      utils.CopyByteSlice(ca.Spec.CheckingKeys[0]),
+		}
+		if len(ca.Spec.SigningKeys) > 0 {
+			kp.PrivateKey = utils.CopyByteSlice(ca.Spec.SigningKeys[0])
+		}
+		keySet.SSH = []*SSHKeyPair{kp}
+	}
+	if len(ca.Spec.TLSKeyPairs) > 0 {
+		keySet.TLS = []*TLSKeyPair{ca.Spec.TLSKeyPairs[0].Clone()}
+	}
+	if len(ca.Spec.JWTKeyPairs) > 0 {
+		keySet.JWT = []*JWTKeyPair{ca.Spec.JWTKeyPairs[0].Clone()}
+	}
+	return keySet
+}
 func (ca *CertAuthorityV2) SetActiveKeys(ks CAKeySet) error {
 	if err := ks.CheckAndSetDefaults(); err != nil {
 		return trace.Wrap(err)
 	}
 	ca.Spec.ActiveKeys = ks
 	return nil
 }
-func (ca *CertAuthorityV2) GetAdditionalTrustedKeys() CAKeySet { return ca.Spec.AdditionalTrustedKeys }
+func (ca *CertAuthorityV2) GetAdditionalTrustedKeys() CAKeySet {
+	return ca.Spec.AdditionalTrustedKeys
```

Comment:
Looks like this return should be removed?

Reply:
right, woops. Should be fixed.